### PR TITLE
Bigquery ADC and Policy Tag implementation

### LINKED ISSUE #2133: [BigQuery] Ingesting BigQuery Policy Tags
## What are BigQuery Policy Tags

It's a tag you assign at per table column level, it is usually to limit the column level access in BigQuery.
See this image
image

## Feature Request

It would be nice to ingest these policy tags into OpenMetadata as column tags here.

image

### LINKED ISSUE #2545: Application Default Credentials for Bigquery
Using Application Default Credentials for Bigquery in order to not use credential files to provide a better and more secure approach to ingest data.

### PR DESCRIPTION
**Pull request by ayush-shah**
### Describe your changes :

Policy Tags and ADC implementation.
fix #2133 & fix #2545 
#
### Type of change :

- [x] Improvement
- [x] New feature

#
### Frontend Preview (Screenshots) :
<p align="center">For frontend related change, please link screenshots of your changes preview! Optional for backend related changes.
</p>

#
### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/open-source-community/developer) document.
- [x] I have performed a self-review of my own. 
- [x] I have tagged my reviewers below.
- [ ] I have commented on my code, particularly in hard-to-understand areas.
- [ ] My changes generate no new warnings.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [x] All new and existing tests passed.

#
### Reviewers

@harshach

### CODE REVIEW

**Comment by ayush-shah** [comment]
Ingestion currently doesn't support creation/updation of tag categories and tags as the Tags currently don't support PUT operations similar to how we have it for Tables, Topics and others.
is it possible for us to have a PUT API created for Tags similarly? 

cc: @harshach @sureshms @pmbrull

**Comment by pmbrull** [comment]
> Ingestion currently doesn't support creation/updation of tag categories and tags as the Tags currently don't support PUT operations similar to how we have it for Tables, Topics and others. is it possible for us to have a PUT API created for Tags similarly?
> 
> cc: @harshach @sureshms @pmbrull

Hi @ayush-shah , for the Tag we have a ticket open to add the proper operations to the python client https://github.com/open-metadata/OpenMetadata/issues/1526

**Comment by ayush-shah** [comment]
> Hi @ayush-shah , for the Tag we have a ticket open to add the proper operations to the python client #1526

Hey @pmbrull, Thanks for the reply, just wanted to let you know that it's a blocker and is dependent on #1526.

File: ingestion/src/metadata/ingestion/ometa/ometa_api.py [hunk 1]
```diff
@@ -503,6 +503,16 @@ def list_tags_by_category(self, category: str) -> List[Tag]:
         resp = self.client.get(f"{self.get_suffix(Tag)}/{category}")
         return [Tag(**d) for d in resp["children"]]
 
+    def create_tag_category(self, data):
+        resp = self.client.post(f"/tags", data=data.json())
+        print(resp)
```
Comment (pmbrull) [inline ingestion/src/metadata/ingestion/ometa/ometa_api.py:508]: Could we remove the prints? 🙏

Comment (ayush-shah) [reply ingestion/src/metadata/ingestion/ometa/ometa_api.py]: Yes, removed those, completely forgot about them.

**Review by pmbrull — APPROVED** [review]
Thanks @ayush-shah, LGTM. Opened https://github.com/open-metadata/OpenMetadata/issues/2623 as a followup

**Pull request merged**